lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -724,6 +725,103 @@ def cmd_services_help(default_columns)
     [ '-h', '--help' ] => [ false, 'Show this help information.' ]
   )
 
+  def db_connection_info(framework)
+    unless framework.db.connection_established?
+      return "#{framework.db.driver} selected, no connection"
+    end
+
+    cdb = ''
+    if framework.db.driver == 'http'
+      cdb = framework.db.name
+    else
+      ::ApplicationRecord.connection_pool.with_connection do |conn|
+        if conn.respond_to?(:current_database)
+          cdb = conn.current_database
+        end
+      end
+    end
+
+    if cdb.empty?
+      output = "Connected Database Name could not be extracted. DB Connection type: #{framework.db.driver}."
+    else
+      output = "Connected to #{cdb}. Connection type: #{framework.db.driver}."
+    end
+
+    output
+  end
+
+  def cmd_db_stats(*args)
+    return unless active?
+    print_line "Session Type: #{db_connection_info(framework)}"
+
+    current_workspace = framework.db.workspace
+    example_workspaces = ::Mdm::Workspace.order(id: :desc)
+    ordered_workspaces = ([current_workspace] + example_workspaces).uniq.sort_by(&:id)
+
+    tbl = Rex::Text::Table.new(
+    'Indent'  => 2,
+    'Header'  => "Database Stats",
+    'Columns' =>
+      [
+        "IsTarget",
+        "ID",
+        "Name",
+        "Hosts",
+        "Services",
+        "Services per Host",
+        "Vulnerabilities",
+        "Vulns per Host",
+        "Notes",
+        "Creds",
+        "Kerberos Cache"
+      ],
+    'SortIndex' => 1,
+    'ColProps' => {
+      'IsTarget' => {
+        'Stylers' => [Msf::Ui::Console::TablePrint::RowIndicatorStyler.new],
+        'ColumnStylers' => [Msf::Ui::Console::TablePrint::OmitColumnHeader.new],
+        'Width' => 2
+      }
+    }
+    )
+    total_creds = 0
+    total_tickets = 0
+    ordered_workspaces.map do |workspace|
+      total_creds += framework.db.creds(workspace: workspace.name).count
+      total_tickets += ticket_search([],nil, workspace).count
```

Comment:
It looks like `[nil]` is required for the `host_ranges` arguments of `ticket_search` to get all the entries. I found this in the `klist` command implementation:

https://github.com/rapid7/metasploit-framework/blob/4065d01c7d3295e9d04943241b489e118f8c8a96/lib/msf/ui/console/command_dispatcher/db/klist.rb#L64-L68

Also, would it be possible to call `ticket_search` just once instead of twice (below line 803)? Maybe using a local variable to store the number of tickets for teach workspace.

```ruby
      nb_tickets = ticket_search([nil],nil, workspace).count
      total_tickets += nb_tickets
      ...
        framework.db.creds(workspace: workspace.name).count.to_fs(:delimited), # workspace.creds.count.to_fs(:delimited) is always 0 for whatever reason
        nb_tickets.to_fs(:delimited)
      ]
```

Obs.: you can use the `admin/kerberos/forge_ticket` module to forge Kerberos ticket easily and populate the database.